### ISSUE TITLE
Improve usability of temporarily disabling strict-blocking

### ISSUE THREAD

**Issue by ArmRes** [uBlockOrigin/uBlock-issues#77]
Now there is an option to turn off the rule, either just now or forever. I would like to be able to disable the rule for a certain time, for example: hour, 6 hours, 12 hours, day, week, month. To do this, you can add a combobox with the selection options next to the "Temporarily" button (I do not know how in the English version).

**Comment by gwarser** [comment]
Define "rule".
Dynamic filtering rules will be active to the end of session if not "commited".

**Comment by gwarser** [comment]
If you mean "options", then duplicate of https://github.com/gorhill/uBlock/issues/2859

**Comment by uBlock-user** [comment]
He wants to disable certain rules with a timer, so not a duplicate.

**Comment by gorhill** [comment]
Unless OP comes up with a better detailed description of what he is suggested, we are not going to guess about it.

**Comment by ArmRes** [comment]
I do not know how to explain it. Add the ability to disable the rules for a specific time, and do this in this lock message.

![screenshot_1](https://user-images.githubusercontent.com/3866363/41512668-cf9596e4-7295-11e8-8aab-9f98c4a86f9a.jpg)

**Comment by uBlock-user** [comment]
That's strict blocking. Do you want to disable strict blocking ?

**Comment by James1o1o** [comment]
I think what OP is suggesting, is that if something is blocked by uBlock Origin, that there be a box that will disable the filter blocking it, but only for a set time, like 15 minutes or so,

**Comment by uBlock-user** [comment]
> that there be a box that will disable the filter blocking it

That's strict blocking and currently it's set to 60 secs for disabling that, which is not configurable, I suppose he wants it to be able to configure to his own liking.

@gorhill what do you think ?

**Comment by gorhill** [comment]
I think the current implementation can be annoying -- disabling strict blocking for 60 s is completely arbitrary. However I won't over-complicate by allowing to select the duration of disabling strict-blocking. A simpler solution is to disable strict-blocking until navigating away from the site, or maybe better, until the tab is closed.

**Comment by uBlock-user** [comment]
So are you declining it ?

**Comment by gorhill** [comment]
Keep open, it's something I agree there is a need for a better behavior than the current arbitrary 60 seconds.

**Comment by uBlock-user** [comment]
You could remove the timer instead, so it will remain whitelisted as long as the user doesn't undo the whitelisting on his own.

**Comment by gorhill** [comment]
There is no whitelisting, the timer is just to not strict-block the page for a while -- it's equivalent to a temporary `no-strict-blocking: example.com true`.

**Comment by gwarser** [comment]
It's more like "Allow once"

**Comment by ArmRes** [comment]
@uBlock-user, yes, to be able to disable temporarily (for a certain period)

**Comment by ArmRes** [comment]
> That's strict blocking and currently it's set to 60 secs for disabling that, which is not configurable, I suppose he wants it to be able to configure to his own liking.

Yes that's right

**Comment by gorhill** [comment]
Three changes:

- Extended the bypass duration to 2 minutes (from 1 minute)
- Added the hidden setting `strictBlockingBypassDuration` (in seconds) to control the bypass duration. Default to 120 seconds.
- The bypass deadline for a given hostname will be pushed to the future (by `strictBlockingBypassDuration` seconds) each time the hostname is visited before the end of the deadline.
